src/persist/src/indexed/columnar/arrow.rs

```diff
@@ -100,7 +100,7 @@ pub(crate) const ENABLE_ARROW_LGALLOC_NONCC_SIZES: Config<bool> = Config::new(
     "A feature flag to enable copying decoded arrow data into lgalloc on non-cc sized clusters.",
 );
 
-fn realloc_data(data: ArrayData, metrics: &ColumnarMetrics) -> ArrayData {
+fn realloc_data(data: ArrayData, nullable: bool, metrics: &ColumnarMetrics) -> ArrayData {
```

Comment:
Totally fine with putting this logic in `realloc_data(...)`, what do you think about renaming the method (and maybe `realloc_array(...)` and `realloc_any(...)`?) something a little more ambiguous though like `reorganize_data(...)`?

Reply:
I've mentioned this in the doc comment, but I'm a bit hesitant to rename the method - a vaguer name feels a bit more confusing to me in other ways, and I'm optimistic enough that we get the bug fixed upstream that I don't want to deform our code around it more than necessary. Can grab this if you feel strongly though!